.github/workflows/diff_chromium_branches.yaml

```diff
@@ -0,0 +1,141 @@
+name: Diff from Chromium branches
+permissions:
+  contents: read
+on:
+  workflow_dispatch:
+    inputs:
+      upstream_branches:
+        description: 'List of Chromium branches to diff from (comma separated).'
+        required: true
+        type: string
+      downstream_branch:
+        description: 'Name of the branch to perform the diff with.'
+        required: true
+        type: string
+  pull_request:
+    types:
+      - labeled
+      - synchronize
+jobs:
+  initialize:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: Set upstream branches
+        id: set-upstream-branches
+        env:
+          PR_LABELS: ${{ toJson(github.event.pull_request.labels) }}
+          EVENT_ACTION: ${{ github.event.action }}
+          UPSTREAM_BRANCHES: ${{ inputs.upstream_branches }}
+        shell: bash
+        run: |
+          set -x
+          if [[ $EVENT_ACTION == "workflow_dispatch" ]]; then
+            # If manually triggered, get branches from comma-separated branches from inputs
+            export DIFF_LABELS=$(echo "$UPSTREAM_BRANCHES" | tr ',' '\n' | sed 's/^[[:space:]]*//;s/[[:space:]]*$//')
+          else
+            # If label or synchronize Get the upstream branches from labels prefixed with $DIFF_PREFIX
+            DIFF_PREFIX=$(cat ${GITHUB_WORKSPACE}/.github/config/chromium_diffing.json | jq -r '.diff_label_prefix')
+            export DIFF_LABELS=$(echo "$PR_LABELS" | jq -r --arg prefix "$DIFF_PREFIX" '.[] | select(.name | startswith($prefix)) | .name | sub($prefix; "")')
+          fi
+          # # Convert into regex needed for filtering with jq
+          SELECTED_BRANCHES=$(echo "$DIFF_LABELS" | tr '\n' '|' | sed 's/|$/\n/')
+          JQ_QUERY="[.stable_branches[] | select(.milestone | test(\"^($SELECTED_BRANCHES)$\"))]"
+          UPSTREAM_BRANCHES=$(cat ${GITHUB_WORKSPACE}/.github/config/chromium_diffing.json | jq "${JQ_QUERY}")
+          echo 'upstream_branches<<EOF' >> $GITHUB_OUTPUT
+          echo "${UPSTREAM_BRANCHES}" >> $GITHUB_OUTPUT
+          echo 'EOF' >> $GITHUB_OUTPUT
+      - name: Set downstream branch
+        id: set-downstream-branch
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        shell: bash
+        run: |
+          set -x
+          if [[ $EVENT_ACTION == "workflow_dispatch" ]]; then
+            # Get the trimmed downstream branch directly from inputs
+            DOWNSTREAM_BRANCH=$(echo ${{ inputs.downstream_branch }} | sed 's/^[[:space:]]*//;s/[[:space:]]*$//)')
+          else
+            # Get the downstream branch from the labelled PR
+            DOWNSTREAM_BRANCH="refs/pull/$PR_NUMBER/head"
```

Comment:
This is accessible directly from the `github` context. See the Workflow Debug run for what's available in there.

https://github.com/youtube/cobalt/actions/runs/15126801841/job/42520216969